File: clients/sdk/frontend/typescript/src/types/thread.ts
export type ThreadType = {
  key: string;
  title?: string;
  created_ts?: number;
  updated_ts?: number;
  author?: string;
  deleted?: boolean;
  kms?: KMSMetaType;
};

export type ThreadCreateRequestType = {
  title: string;
};

export type ThreadUpdateRequestType = {
  title?: string;
};

export type ThreadResponseType = {
  thread: ThreadType;
};

export type KMSMetaType = {
  key_id?: string;
  wrapped_dek?: string;
  kek_id?: string;
  kek_version?: string;
};

File: clients/sdk/backend/python/src/progressdb/types/thread.py
from typing import TypedDict, Optional


class ThreadType(TypedDict):
    key: str
    title: Optional[str]
    created_ts: Optional[int]
    updated_ts: Optional[int]
    author: str
    deleted: Optional[bool]
    kms: Optional[dict]  # KMSMetaType


class ThreadCreateRequestType(TypedDict):
    title: str


class ThreadUpdateRequestType(TypedDict, total=False):
    title: str